test/resource/custom-cluster/custom-xml-in-zcl-json.xml

```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0"?>
+<!--
+Copyright (c) 2025 Silicon Labs.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<configurator>
+  <domain name="CHIP"/>
+
+  <!--
+    These are test global items (no cluster attached) for testing only.
+    Their usage is defined for UnitTestCluster only.
+  -->
+  <clusterExtension code="0x0201">
+    <struct name="AdditionalInfoStruct">
```

Comment:
We should make this work for structs at toplevel, not inside `<clusterExtension>`, for two reasons:

1. Structs can be referenced by more than one cluster and hence more than one clusterExtension.  Placing the struct inside one of them would then potentially break the other, depending on processing order.
2. Even when not doing clusterExtension and just defining data types, shouldn't we support a struct in file A referencing a data type defined in file B, just like we support a command or an event or an attribute in file A referencing a type defined in file B?

Reply:
Data types need not be within the clusterExtensions.